homeassistant/components/life360/config_flow.py

```diff
@@ -1,108 +1,251 @@
+                self._first_reauth_confirm = False
+                schema = password_schema(pwd)
+            return self.async_show_form(
+                step_id=step_id, data_schema=vol.Schema(schema), errors=errors
+            )
 
-        return self.async_show_form(
-            step_id="user",
-            data_schema=vol.Schema(data_schema),
-            errors=errors,
-            description_placeholders={"docs_url": DOCS_URL},
+        data = {
+            CONF_USERNAME: self._username,
+            CONF_PASSWORD: self._password,
+            CONF_AUTHORIZATION: authorization,
+        }
+
+        if self._reauth_entry:
+            LOGGER.info("Reauthorization successful")
+            self.hass.config_entries.async_update_entry(self._reauth_entry, data=data)
+            if self._reauth_entry.state == ConfigEntryState.LOADED:
+                # Config entry reload should not be necessary. Restarting coordinator's
+                # scheduled refreshes should be sufficient since Life360 api object is
+                # valid again after successful reauthorization.
+                coordinator = self.hass.data[DOMAIN]["accounts"][self.unique_id][
+                    "coordinator"
+                ]
+                self.hass.async_create_task(coordinator.async_request_refresh())
+            else:
+                # Config entry never got completely loaded, so do a full reload.
+                self.hass.async_create_task(
+                    self.hass.config_entries.async_reload(self._reauth_entry.entry_id)
+                )
+            return self.async_abort(reason="reauth_successful")
+
+        init_integ_data(self.hass)
+        self.hass.data[DOMAIN]["accounts"][cast(str, self.unique_id)] = AccountData(
+            api=self._api
+        )
+        return self.async_create_entry(
+            title=cast(str, self.unique_id), data=data, options=self._options
         )
 
-    async def async_step_import(self, user_input):
-        """Import a config flow from configuration."""
-        username = user_input[CONF_USERNAME]
-        password = user_input[CONF_PASSWORD]
-        try:
-            authorization = await self.hass.async_add_executor_job(
-                self._api.get_authorization, username, password
-            )
-        except LoginError:
-            _LOGGER.error("Invalid credentials for %s", username)
-            return self.async_abort(reason="invalid_auth")
-        except Life360Error as error:
-            _LOGGER.error(
-                "Unexpected error communicating with Life360 server: %s", error
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a config flow initiated by the user."""
+        if not user_input:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema(
+                    account_schema() | _account_options_schema(self._options)
```

Comment:
Only items that are required for the authentication and connection are allowed in the main flow. Optional customizations should only be part of the options flow.

Reply:
That means that the entries will be created potentially with different options than the user wants, meaning they will behave differently than expected for the time it takes the user to bring up the options and change them. Not optimal in my opinion, but if that's what you want, ok.

Reply:
We aim for setting sane defaults for optional options and let the user change those if they want after the initial setup

Reply:
Ok